### DISCUSSION TITLE
Conhecimentos essenciais

### DISCUSSION THREAD

**Discussion by josejuniordev** [frontendbr/forum#1401] [Dicas]
Bora ajudar a comunidade??

Amigos, boa tarde!

Se eu postei no lugar errado, me perdoem e podem apagar, mas resolvi postar aqui pois aqui também tem pessoas mais experientes que poderão contribuir e muito.

Em algumas empresas que trabalhei eu já participei de processos de recrutamento e diversas vezes percebi que muitos profissionais tem défice de conhecimento.

Esse gap, pode ser por causa do background de cada um. Por exemplo: um dev front-end (ou back-end) que venha do design pode não ter todos os conhecimentos que um outro que tenha cursado ciências da computação tem, e vice versa, como Complexidade de algoritmos e Teoria das cores (ou semiótica).

Até mesmo não importa se fez ou não uma faculdade, pois, existem muitos devs que estudaram e aprenderam sozinhos, na raça, 'fuçando'...

A questão é: se uma pessoa não sabe que precisa de determinado conhecimento, ela nunca irá buscar estudar aquilo, pois, nem sabe que existia tal matéria.

Pronto! Chegamos no problema.

Pensando nisso eu resolvi criar um repositório colaborativo onde quem quiser contribuir poderá adicionar ou remover conteúdos, que servirão de guia para que TODOS possam estudar coisas que talvez nem saibam que existam.

A ideia é ter neste repositório links para conteúdos (posts, videos, pdfs, etc) que vocês da comunidade pensam ser indispensáveis para qualquer pessoa da área.

Quem sabe assim consigamos nivelar os conhecimentos? Obviamente, cada um na sua área de atuação.
Sei que tem devs que gostariam ou precisam de conhecimentos sobre Design pois querem fazer os próprios layouts dos freelas que pegam. E também existem devs que não conhecem certas materias teóricas da programação, pois não cursaram uma faculdade e não chegaram a ter conhecimento delas sozinhos.

Enfim, assuntos sobre Design, Desenvolvimento, DevOps... e tudo mais que pensarem ser importante.

[https://github.com/josejuniordev/essential-knowledge](https://github.com/josejuniordev/essential-knowledge)

Por favor, quem quiser contribuir é mais que bem vindo. É só abrir as PR's.

Bora ajudar a comunidade!

**Comment by jair-anderson-souza** [comment]
Dá uma sacada nesse repo @josejuniordev: https://github.com/kamranahmedse/developer-roadmap

**Comment by josejuniordev** [comment]
@jass2125
sim muito top esse repo. Eu já o conhecia. Mas bem lembrado! vai agregar.
Mas a ideia é ir um pouco mais profundo também.
Tipo: sabe aquele conhecimento que vc talvez tenha tido acesso apenas por causa da faculdade e que vc pensa ser crucial pra todo dev saber, porém, nem todo mundo vai ter acesso ou ciencia de que existe por talvez ter feito outra facul ou até mesmo nem ter feito?

Esses são importantes para o repo tbm!!

o que acham?

**Comment by josejuniordev** [comment]
@NicholasKuchiniski 
Concordo com vc em que não é necessário ter ensino superior pra nossa área, tanto que o texto diz:

> Até mesmo não importa se fez ou não uma faculdade, pois, existem muitos devs que estudaram e aprenderam sozinhos, na raça, 'fuçando'...
A questão é: se uma pessoa não sabe que precisa de determinado conhecimento, ela nunca irá buscar estudar aquilo, pois, nem sabe que existia tal matéria

Embora algumas pessoas optem por fazer, apenas para "ter um diploma" sei lá... No seu caso, qual tem sido a motivação em estar cursando uma?

Conheço alguns programadores **muito bons** e eles não tem ensino superior.

Outro ponto, eu mesmo, já trabalho há mais de 13 anos e não cursei 'faculdade de programação' também.

A idéia pra esse repositório veio depois que ví 2 vagas onde os entrevistados foram questionados justamente sobre qual era a 'complexidade do algorítmo' que elas haviam acabado de escrever, e alguns não souberam responder, pois, nem conheciam esse assunto.

E cá entre nós... este assunto cobrado é algo realmente imprescindível? Talvez pro negocio da empresa que estava contratando fosse. Talvez realmente seja... será que é?

Outro exemplo: conheço vários desenvolvedores front-end que são bem ruins de semântica, acessibilidade... E **no meu ponto de vista**, pra ser um desenvolvedor front-end completo você precisa conhecer o mínimo sobre esses assuntos entre outros, senão você é apenas um programador que estará atuando com javascript, com ou sem libs e frameworks.
E o curioso é que conversando com alguns deles, estes nem sabiam que talvez devessem estudar essas coisas.

Enfim, o ponto aqui não é o ensino superior, não é fornecer no repo TODOS os conteúdos que o ensino superior fornece. **O ponto aqui é o conhecimento**, listar todos os assuntos que realmente são importantes, inclusive teóricos e já com links pra bons conteúdos, que poderão fazer com que uma determinada pessoa seja um profissional melhor.
Ajudar, direcionar e dar um norte pra galera independentemente deles fazerem uma faculdade ou não, afinal de contas **sempre vai depender do esforço individual** de cada um.
